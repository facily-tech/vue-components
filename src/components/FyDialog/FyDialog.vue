<template>
  <v-dialog
    v-model="dialog"
    persistent
    :fullscreen="config.fullscreen"
    :hide-overlay="config.hideOverlay"
    :scrollable="config.scrollable"
    v-bind="propsAndAttrs"
    v-on="$listeners"
  >
    <slot v-for="(_, name) in $slots" :name="name" :slot="name" />
    <slot name="content"></slot>
  </v-dialog>
</template>

<script lang="ts">
import Vue from 'vue';
import { IDialog } from './types';

export default Vue.extend({
  name: 'fy-dialog',

  props: {
    config: {
      type: Object as () => IDialog,
      default: () => ({ fullscreen: false, hideOverlay: false, scrollable: false } as IDialog),
    },
    dialog: { type: Boolean, default: false },
  },

  computed: {
    propsAndAttrs(): Record<string, unknown> {
      return {
        ...this.$attrs,
        ...this.$props,
      };
    },
  },
});
</script>
