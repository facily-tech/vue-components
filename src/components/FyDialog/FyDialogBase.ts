/// <reference path="../../../node_modules/vuetify/src/globals.d.ts" />
import Vue from 'vue';

import { VDialog } from 'vuetify/lib';

const BaseSnackbar = Vue.extend({ extends: VDialog, mixins: [VDialog] });

type options = InstanceType<typeof BaseSnackbar>;

export default BaseSnackbar.extend<options>().extend({
  name: 'fy-dialog-base',
  computed: {
    class() {
      return {
        ...VDialog.options.computed.classes.call(this),
        'fy-dialog-base': true,
      };
    },
  },
});
