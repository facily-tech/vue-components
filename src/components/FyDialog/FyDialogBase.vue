<template>
  <v-dialog v-bind="$attrs" v-on="$listeners" />
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'FyDialogBase',
});
</script>
