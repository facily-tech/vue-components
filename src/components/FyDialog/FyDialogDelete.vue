<template>
  <fy-dialog
    :dialog="deleteModel"
    :config="{
      fullscreen: false,
      hideOverlay: false,
      scrollable: true,
    }"
    :max-width="'580px'"
  >
    <template v-slot:content>
      <v-card>
        <div class="delete-wrapper" data-test="dialog-delete">
          <v-card tile>
            <v-toolbar flat light color="primary">
              <v-toolbar-title style="color: white">
                {{ title }}
              </v-toolbar-title>
              <v-spacer></v-spacer>
              <fy-button icon @click="close()">
                <v-icon color="#ffffff"> mdi-close </v-icon>
              </fy-button>
            </v-toolbar>
            <v-card-text class="pa-10">
              <div class="dtext-center pt-5">
                <slot name="content">
                  <h2 class="headline">
                    Tem certeza que deseja excluir <strong>"{{ item.description }}"</strong>?
                  </h2>
                </slot>
              </div>

              <div class="justify-center d-flex mt-10">
                <fy-button
                  color="primary"
                  :loading="loading"
                  large
                  @click="submit()"
                  data-test="btn-submit"
                  class="ml-3"
                >
                  Excluir
                </fy-button>
                <fy-button large @click="close()" data-test="btn-close" class="ml-3">
                  Cancelar
                </fy-button>
              </div>
            </v-card-text>
          </v-card>
        </div>
      </v-card>
    </template>
  </fy-dialog>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';

import { FyDialog } from './index';

import FyButton from '@/components/FyButton';

import { IDialogDeleteItem } from './types';

export default Vue.extend({
  name: 'fy-dialog-delete',

  components: { FyButton, FyDialog },

  props: {
    item: {
      type: Object as PropType<IDialogDeleteItem>,
      default: {} as IDialogDeleteItem,
      required: true,
    },

    deleteModel: { type: Boolean, default: false, required: true },

    title: { type: String, default: 'Excluir' },
  },

  data() {
    return {
      loading: false,
    };
  },

  methods: {
    close(): void {
      this.$emit('close-dialog');
    },

    async submit(): Promise<void> {
      this.$emit('close-dialog');
      this.$emit('confirm-dialog', this.item);
    },
  },
});
</script>
