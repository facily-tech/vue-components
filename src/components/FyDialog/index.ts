export { default as FyDialog } from './FyDialog.vue';
export { default as FyDialogDelete } from './FyDialogDelete.vue';
export { default as FyDialogBase } from './FyDialogBase.vue';
