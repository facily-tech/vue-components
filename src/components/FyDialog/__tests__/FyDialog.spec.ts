import Vue from 'vue';
import Vuetify from 'vuetify';
import FyDialog from '../FyDialog.vue';
import { mount, MountOptions, Wrapper } from '@vue/test-utils';

Vue.use(Vuetify);

describe('FyDialog.ts', () => {
  // eslint-disable-line max-statements
  type Instance = InstanceType<typeof FyDialog>;
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(FyDialog, {
        vuetify: new Vuetify(),
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
