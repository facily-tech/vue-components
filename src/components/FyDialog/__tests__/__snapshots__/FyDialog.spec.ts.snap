// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyDialog.ts should render component and match snapshot 1`] = `
<div class="v-dialog__container" config="[object Object]">
  <!---->
</div>
`;
